### PR TITLE
config: tgl-cavs: add kpb, selector and kd support

### PR DESCRIPTION
**Pull request by libinyang**
Add KPB, Selector and KD module config

Signed-off-by: Libin Yang <libin.yang@intel.com>

### CODE REVIEW

File: config/tgl-cavs.toml [hunk 1]
```diff
@@ -266,3 +266,58 @@ count = 9
 		20, 0, 0, 0, 12832, 20881000, 0, 0, 0, 20881, 0,
 		21, 0, 0, 0, 12832, 23431000, 0, 0, 0, 23431, 0,
 		22, 0, 0, 0, 12832, 30471000, 0, 0, 0, 30471, 0]
+
```
Comment (plbossart) [inline config/tgl-cavs.toml:269]: thsi conflicts with https://github.com/thesofproject/rimage/pull/120

@libinyang and @aiChaoSONG can't you guys work together?

**Review by plbossart — CHANGES_REQUESTED** [review]

File: config/tgl-cavs.toml (hunk 1 above)
Comment (libinyang) [reply config/tgl-cavs.toml:269]: @plbossart I will refine the patch after https://github.com/thesofproject/rimage/pull/120 is merged.

**Comment by libinyang** [comment]
patch update based on the latest code. 
This PR conflicts with https://github.com/thesofproject/rimage/pull/124. Whichever PR is merged, the other PR should be rebased.

**Comment by lgirdwood** [comment]
@libinyang #124 now merged, so will need a rebase.

**Comment by libinyang** [comment]
patch updated based on the latest code.

**Comment by libinyang** [comment]
One question unrelated to this PR: I found that the module_type IDs for each module in tgl are not exactly same as the IDs in mtl. Do we have plan to make them be the same, which is suggested in https://github.com/thesofproject/rimage/pull/117/commits/9802166b7c0f8430298c02736658472ca528d5c3?

**Comment by lgirdwood** [comment]
@RanderWang @libinyang can we align to modules types in MTL. i.e. fix the TGL types. @ujfalusi @ranj063 any kernel impact ?

**Comment by RanderWang** [comment]
> 

we need to align it to  the following definition which are defined by latest windows driver.  This type is only used by windows driver. Linux driver uses uuid to identify module
````
enum mod_type {
        ebasefw = 0,
        emixin,
        emixout,
        ecopier,
        epeakvol,
        eupdwmix,
        emux,
        esrc,
        ewov,
        efx,
        eaec,
        ekpb,
        emicselect,
        efxf,   /*i.e.SmartAmp */
        eaudclass,
        efakecopier,
        eiodriver,
        ewhm,
        egdbstub,
        esensing,
        emax,
        einvalid = emax
} ;
````

**Comment by libinyang** [comment]
PR updated
Chao has submitted a patch to align the module_types. update the patch based on the latest code

**Review by RanderWang — APPROVED** [review]

**Comment by kv2019i** [comment]
@RanderWang @lgirdwood I don't think the module_type now has impact, but we definitely need to write down the semantics , e.g. in https://github.com/thesofproject/sof-docs/pull/451/commits/3209a52636e8a20082f4354e7dc45bf6305f4d38

**Comment by lgirdwood** [comment]
@plbossart ping ?

**Comment by libinyang** [comment]
> @plbossart ping ?

@lgirdwood As @kai mentioned, it seems module_type doesn't have impact currently and my test supports Kai's conclusion. Anyway, I think we should set the module_type correctly in rImage in case we enable the module_type later to align with Windows behavior.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**